app/src/main/java/com/mapbox/maps/testapp/examples/LocationTrackingActivity.kt

```diff
@@ -3,77 +3,115 @@ package com.mapbox.maps.testapp.examples
  */
-class LocationTrackingActivity : AppCompatActivity(), OnCameraTrackingChangedListener {
+class LocationTrackingActivity : AppCompatActivity() {
 
   private lateinit var locationPermissionHelper: LocationPermissionHelper
-  private lateinit var mapboxMap: MapboxMap
+
+  private val onIndicatorBearingChangedListener = OnIndicatorBearingChangedListener {
+    mapView.getMapboxMap().jumpTo(CameraOptions.Builder().bearing(it).build())
+  }
+
+  private val onIndicatorPositionChangedListener = OnIndicatorPositionChangedListener {
+    mapView.getMapboxMap().jumpTo(CameraOptions.Builder().center(it).build())
+    mapView.getGesturesPlugin().focalPoint = mapView.getMapboxMap().pixelForCoordinate(it)
+  }
+
+  private val onMoveListener = object : OnMoveListener {
+    override fun onMoveBegin(detector: MoveGestureDetector) {
+      onCameraTrackingDismissed()
+    }
+
+    override fun onMove(detector: MoveGestureDetector): Boolean {
+      return false
+    }
+
+    override fun onMoveEnd(detector: MoveGestureDetector) {}
+  }
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_simple_map)
-    mapboxMap = mapView.getMapboxMap()
     locationPermissionHelper = LocationPermissionHelper(this)
     locationPermissionHelper.checkPermissions {
       onMapReady()
     }
   }
 
   private fun onMapReady() {
-    mapboxMap.jumpTo(
+    mapView.getMapboxMap().jumpTo(
       CameraOptions.Builder()
         .zoom(14.0)
         .build()
     )
-    mapboxMap.loadStyleUri(
+    mapView.getMapboxMap().loadStyleUri(
       Style.MAPBOX_STREETS
     ) {
-      initLocationComponent(it)
-      Style.MAPBOX_STREETS
+      initLocationComponent()
+      setupGesturesListener()
     }
   }
 
-  private fun initLocationComponent(style: Style) {
-    val locationPluginImpl = mapView.getLocationPlugin()
-    locationPluginImpl.activateLocationComponent(
-      LocationComponentActivationOptions
-        .builder(this, style)
-        .useDefaultLocationEngine(true)
-        .locationEngineRequest(
-          LocationEngineRequest.Builder(750)
-            .setFastestInterval(750)
-            .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
-            .build()
-        )
-        .build()
-    )
-    locationPluginImpl.addOnCameraTrackingChangedListener(this)
-    locationPluginImpl.cameraMode = CameraMode.TRACKING_GPS // CameraMode.TRACKING_GPS_NORTH
-    locationPluginImpl.renderMode = RenderMode.GPS
-    locationPluginImpl.enabled = true
+  private fun setupGesturesListener() {
+    mapView.getGesturesPlugin().addOnMoveListener(onMoveListener)
   }
 
-  override fun onCameraTrackingDismissed() {
-    Toast.makeText(this, "onCameraTrackingDismissed", Toast.LENGTH_SHORT).show()
+  private fun initLocationComponent() {
+    val locationComponentPlugin = mapView.getLocationComponentPlugin()
+    locationComponentPlugin.updateSettings {
+      this.enabled = true
```

Comment:
So do I get it correctly that if we miss that line - nothing will work? 😄

Reply:
right, we have to enable the location component plugin, we can either do it here in the code, or through adding the xml attribute.

Reply:
Can we try enabling plugin (if it's not enabled) if any function (like `updateSettings` in this case) invoked?

Reply:
I think the setting properties themselves are pretty clear, we should avoid adding too much logic and confuse users.